src/utils/context_free_grammar_generator.cpp

```diff
@@ -125,36 +124,60 @@ context_free_grammar_generator::context_free_grammar_generator(const std::map<st
 			}
 		}
 	}
-	initialized_ = true;
 }
 
-std::string context_free_grammar_generator::print_nonterminal(const std::string& name, uint32_t* seed, short seed_pos) const {
-	std::string result;
-	std::map<std::string, nonterminal>::const_iterator found = nonterminals_.find(name);
-	if (found == nonterminals_.end()) {
-		lg::wml_error() << "[context_free_grammar_generator] Warning: needed nonterminal " << name << " not defined";
-		return "!" + name;
+std::string context_free_grammar_generator::print_nonterminal(const std::string& name, const std::map<std::string,std::string>* variables, uint32_t* seed, short seed_pos) const {
+	if (name.length() >= 2 && name[0] == '$') {
+		std::string variable = name.substr(1);
+		if (variables != nullptr) {
+			std::map<std::string,std::string>::const_iterator found = variables->find(variable);
+			if (found != variables->end()) {
+				return found->second;
+			}
+		}
+
+		lg::wml_error() << "[context_free_grammar_generator] Warning: needed variable" << variable << " not defined";
+		return "!" + variable;
 	}
-	const context_free_grammar_generator::nonterminal& got = found->second;
-	unsigned int picked = seed[seed_pos++] % got.possibilities_.size();
-	if (seed_pos >= seed_size) seed_pos = 0;
-	if (picked == got.last_) {
-		picked = seed[seed_pos++] % got.possibilities_.size();
+	else {
```

Comment:
...you know, this else isn't actually needed since the if block always returns. It would make the diff clearer, as well.